spec/util/miq-xml_spec.rb

```diff
@@ -6,15 +6,38 @@
     expect(xml.root.elements[1].attributes['attr1']).to eq(attr_string)
   end
 
+  it "handles loaded document with top-level text nodes" do
+    attr_string = "test string"
+    doc_text = "XXX<test><element_1 attr1='#{attr_string}'/></test>"
+
+    xml = MiqXml.load(doc_text)
+    expect(xml.root.elements[1].attributes['attr1']).to eq(attr_string)
+
+    expect(xml.to_s).to start_with("XXX<test>")
+
+    xml.write(xml_str = '', 1)
+    if REXML::VERSION < "3.3.2"
+      expect(xml_str).to start_with("XXX\n<test>")
+    else
+      expect(xml_str).to start_with("\n<test>")
+    end
```

Comment:
I'm not sure if we even need this part of the test - the real test is ensuring we can load the document and find the attribute (lines 13-14).

So, I kept it and used a version conditional here, which is gross, but I'm not sure I care too much.

Alternatively, we can _force_ the minimum version of REXML in the gemspec.  There's a CVE anyway, so we may want to do that.  However, when I tried to do that locally I was getting bundler errors and I'm not sure why.

Reply:
Actually, now that I wrote that - I'm gonna change the .gemspec anyway to set a minimum, because of the CVE.